From 2e26af756d69c53026bf2d6d61589f754796340d Mon Sep 17 00:00:00 2001
From: Paul Floyd <pjfloyd@wanadoo.fr>
Date: Sat, 27 Apr 2024 13:07:07 +0200
Subject: [PATCH 02/11] 486180 [MIPS] 'VexGuestArchState' has no member named
 'guest_IP_AT_SYSCALL'

(cherry picked from commit 7214886886bce9029f325214156c02dcfff760d5)
---
 NEWS                          | 2 ++
 VEX/priv/guest_mips_helpers.c | 4 ++++
 VEX/pub/libvex_guest_mips32.h | 6 +++---
 VEX/pub/libvex_guest_mips64.h | 2 ++
 4 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/NEWS b/NEWS
index 8bdd3753f7bb..c40e00cce46b 100644
--- a/NEWS
+++ b/NEWS
@@ -5,6 +5,8 @@ Branch 3.23
 
 The following bugs have been fixed or resolved on this branch.
 
+486180  [MIPS] 'VexGuestArchState' has no member named 'guest_IP_AT_SYSCALL'
+
 To see details of a given bug, visit
   https://bugs.kde.org/show_bug.cgi?id=XXXXXX
 where XXXXXX is the bug number as listed above.
diff --git a/VEX/priv/guest_mips_helpers.c b/VEX/priv/guest_mips_helpers.c
index 74cfb9c34caa..79197378cc74 100644
--- a/VEX/priv/guest_mips_helpers.c
+++ b/VEX/priv/guest_mips_helpers.c
@@ -187,6 +187,8 @@ void LibVEX_GuestMIPS32_initialise( /*OUT*/ VexGuestMIPS32State * vex_state)
    vex_state->guest_w1.w64[1] = 0;
    vex_state->guest_w2.w64[0] = 0;
    vex_state->guest_w2.w64[1] = 0;
+
+   vex_state->guest_IP_AT_SYSCALL = 0;
 }
 
 void LibVEX_GuestMIPS64_initialise ( /*OUT*/ VexGuestMIPS64State * vex_state )
@@ -294,6 +296,8 @@ void LibVEX_GuestMIPS64_initialise ( /*OUT*/ VexGuestMIPS64State * vex_state )
    vex_state->guest_LLaddr = 0xFFFFFFFFFFFFFFFFULL;
    vex_state->guest_LLdata = 0;
 
+   vex_state->guest_IP_AT_SYSCALL = 0;
+
    vex_state->guest_MSACSR = 0;
 }
 
diff --git a/VEX/pub/libvex_guest_mips32.h b/VEX/pub/libvex_guest_mips32.h
index 214f63cdbe8c..e769819d75ae 100644
--- a/VEX/pub/libvex_guest_mips32.h
+++ b/VEX/pub/libvex_guest_mips32.h
@@ -188,10 +188,10 @@ typedef
 
       /*  1016 */ UInt guest_MSACSR;
 
-      /*  1020 */ UInt _padding3;
+      /*  1020 */ UInt guest_IP_AT_SYSCALL;
 
-      /*  1020 */ ULong guest_LLdata64;
-      /*  1028 */ ULong _padding4;
+      /*  1024 */ ULong guest_LLdata64;
+      /*  1032 */ ULong _padding3;
 } VexGuestMIPS32State;
 /*---------------------------------------------------------------*/
 /*--- Utility functions for MIPS32 guest stuff.               ---*/
diff --git a/VEX/pub/libvex_guest_mips64.h b/VEX/pub/libvex_guest_mips64.h
index 657fe6fa3343..a953f0ab86a2 100644
--- a/VEX/pub/libvex_guest_mips64.h
+++ b/VEX/pub/libvex_guest_mips64.h
@@ -184,6 +184,8 @@ typedef
       /* 1144 */ UInt guest_MSACSR;
 
       /* 1148 */ UInt _padding2;
+      /* 1152 */ ULong guest_IP_AT_SYSCALL;
+      /* 1160 */ ULong _padding3;
 
 } VexGuestMIPS64State;
 
-- 
2.45.2

